pkg/envoy/agent.go

```diff
@@ -148,17 +148,22 @@ func (a *Agent) terminate() {
 		time.Sleep(a.minDrainDuration)
 		log.Infof("Checking for active connections...")
 		ticker := time.NewTicker(activeConnectionCheckDelay)
-		for range ticker.C {
+		select {
+		case <-ticker.C:
 			if a.activeProxyConnections() == 0 {
 				log.Info("There are no more active connections. terminating proxy...")
 				a.abortCh <- errAbort
 				return
 			}
+		case <-time.After(a.terminationDrainDuration):
```

Comment:
With defaults of minDrainDuration=5s and terminationDrainDuration=5s, this means the activeProxyConnections is effectively never used...?

It seems like this value should equal to terminationGracePeriodSeconds on pod spec (even then its not *really* required, since k8s will force kill it at that point)?

Reply:
Oh. That is a good point. I did not think about defaults. It makes sense to equal it to terminationGracePeriodSeconds. But in our environment k8s never killed even after 20 mins

Reply:
looks like terminationGracePeriodSeconds is available only in 1.20 - that may be the reason why k8s did not kill our pod.  On 1.20 looks like k8s might kill it - So should we just ignore this for now?

Reply:
Where did you see that? I think its wrong. I saw it on https://sysdig.com/blog/kubernetes-1-21-whats-new/ but its not accurate, the new feature is node graceful shutdown, not pod, AFAIK

Reply:
Do not remember where I have seen. Let me see if I can get the link. Let me also figure out why k8s never killed the pod - btw it is a job not sure if it matters.